### DISCUSSION TITLE
Help with understanding JavaScript editor in Connect

### DISCUSSION THREAD

**Discussion by jtyme** [nextgenhealthcare/connect#5873] [Q&A] [answered]
Fairly new to Mirth and Java, all my development life has been in C# so trying to grasp Rhino and Javascript in Mirth Connect. Trying to understand how JavaScript and Java are used inside JavaScript editor. Not been having much luck in the world of Google finding clear answers.

Working with strings I am trying to understand what each syntax represents.  So first off, is the JavaScript editor a Rhino Editor? From what I am researching, Rhino is a project that allows JavaScript syntax to work inside Java applications. Does it take JavaScript syntax and convert it to Java to be used via Java classes? 

I know doing var s = "string literal" and that var s = new String('Object') and that var s = 'string' + '' is equivalent to var s = String('string') which is primitive. But all all three different versions of a class Java.lang.string or is some of them a string.prototype (javascript string).  I am getting confused when a value is based on JavaScript and others are based on Java. 

Thanks for any light or good articles that discusses this information.

**Comment by jonbartels** [comment, marked as answer]
> From what I am researching, Rhino is a project that allows JavaScript syntax to work inside Java applications

This is correct. The way its implemented in Mirth is that you write Javascript, which can call Java classes, in your transformers then those scripts are executable in Java.

The string, String, String problem you describe is a common headache. I am not able to explain it though. You get used to it.

Another big item to consider for Mirth is that it loads the older e4x (ECMA 4 XML) library and is how MC handles XML. E4X is powerful but a bit dated.
